### DISCUSSION TITLE
Followers / creatures willing to join logic

### DISCUSSION THREAD

**Discussion by a1exsh** [ihhub/fheroes2#4490] [General]
Already when playing the original game I often found decisions of the map monsters to join the army of my heroes quite odd, for example:
- Elves eagerly joining all undead troops
- Creatures of the same kind that a hero already has in the army don't seem to be joining more eagerly than any others
- Creatures of the same race also seem to join only randomly

Did anyone else also notice the same?

Additionally, I think it'd make sense if the Leadership skill improved the chances of wandering neutrals joining you. ;)

**Comment by Branikolog** [comment]
Hi, @a1exsh !
Talking about the chance of joining creatures according to their type - not sure this is an original behaviour. I hope our Original Game Specialist @LeHerosInconnu would tell whether chances are affected by this. :)



> Additionally, I think it'd make sense if the Leadership skill improved the chances of wandering neutrals joining you. ;)

It's kind of mod-like feature. (Not bad, actually) Changing the existing secondary skills' features is not the optimal way of improving the game. This is not friendly to old players, who got used to the original leadership characteristic. 
Anyway, such changes could be a part of some sort of "polishing" mode in future. ;)

**Reply by LeHerosInconnu** [reply]
Hello @a1exsh and @Branikolog,

> Hi, @a1exsh ! Talking about the chance of joining creatures according to their type - not sure this is an original behaviour. I hope our Original Game Specialist @LeHerosInconnu would tell whether chances are affected by this. :)

This appeared in Heroes 3.
In Heroes 2, only the power of the army is taken into account.
For a wandering troop to join the hero, the army must be at least more than twice as strong and not more than five times as strong as the power of the wandering troop.
Note, that there is a bug in the original game (or an unknown rule, but I don't see why) and that it doesn't always work as expected (especially when the army strength is more than five times the wandering troop strength).

> > Additionally, I think it'd make sense if the Leadership skill improved the chances of wandering neutrals joining you. ;)
> 
> It's kind of mod-like feature. (Not bad, actually) Changing the existing secondary skills' features is not the optimal way of improving the game. This is not friendly to old players, who got used to the original leadership characteristic. Anyway, such changes could be a part of some sort of "polishing" mode in future. ;)

Yes, this could be integrated eventually as an option.
Personally, I wouldn't use it, I prefer the simplicity of Heroes 2, and there are also fewer slots in the hero's army in Heroes 2. :)
